### PR TITLE
Formula component dynamic category selects

### PR DESCRIPTION
**Pull request by vrackovic**
- Category fields in formula components can be added or removed dynamically
- Implemented react-select
- Slightly changed styles

### CODE REVIEW

File: web/src/main/frontend/src/components/FormulaItem.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    const {id, changeCategory} = this.props;
+    this.setState({
+      selectedCategory: selection.value
+    });
+    changeCategory({...selection, id});
+  };
+
+  render() {
+    return (
+      <div className="formula__category_wrapper" key={this.props.id}>
```
Comment (iscekic) [inline web/src/main/frontend/src/components/FormulaItem.js]: Why do we need `key` here?

File: web/src/main/frontend/src/components/FormulaItem.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+@inject('formula')
+@observer
+class FormulaItem extends React.Component {
+  state = {
+    selectedCategory: ''
+  };
+
+  handleCategoryChange = selection => {
+    const {id, changeCategory} = this.props;
+    this.setState({
```
Comment (iscekic) [inline web/src/main/frontend/src/components/FormulaItem.js]: Can we remove local state?

File: web/src/main/frontend/src/components/FormulaItem.js (hunk 1 above)
Comment (vrackovic) [reply web/src/main/frontend/src/components/FormulaItem.js]: We don't :)

File: web/src/main/frontend/src/components/FormulaItem.js (hunk 2 above)
Comment (vrackovic) [reply web/src/main/frontend/src/components/FormulaItem.js]: We can try :)

File: web/src/main/frontend/src/base/stores/FormulaStore.js [hunk 3]
```diff
@@ -1,12 +1,37 @@
...
+      this.formCategories = this.formCategories.filter(
+        category => category.id !== categoryId
+      );
+    }
+  }
+
+  @action.bound
+  setFormCategories(category) {
+    this.formCategories = this.formCategories.map(
+      form => (form.id === category.id ? category : form)
```
Comment (iscekic) [inline web/src/main/frontend/src/base/stores/FormulaStore.js]: Rename `form` to `existingCategory`.

**Pull request merged**